src/fetch.ts

```diff
@@ -31,45 +31,64 @@ async function fetchLinkedStylesheets(
   );
 }
 
+const ELEMENTS_WITH_INLINE_ANCHOR_STYLES_QUERY = '[style*="anchor"]';
 // Searches for all elements with inline style attributes that include `anchor`.
 // For each element found, adds a new 'data-has-inline-styles' attribute with a
 // random UUID value, and then formats the styles in the same manner as CSS from
 // style tags.
-function fetchInlineStyles() {
-  const elementsWithInlineAnchorStyles: NodeListOf<HTMLElement> =
-    document.querySelectorAll('[style*="anchor"]');
+function fetchInlineStyles(elements?: HTMLElement[]) {
+  const elementsWithInlineAnchorStyles: HTMLElement[] = elements
+    ? elements.filter((el) =>
+        el?.matches?.(ELEMENTS_WITH_INLINE_ANCHOR_STYLES_QUERY),
+      )
```

Comment:
It doesn't hurt anything, but I'm curious why the optional chaining is needed here.

```suggestion
    ? elements.filter((el) =>
        el.matches(ELEMENTS_WITH_INLINE_ANCHOR_STYLES_QUERY),
      )
```

Reply:
I think I probably used them before I added type checking to make sure they will indeed be `HTMLElement`s. I'll remove these.